dspace-api/src/main/java/org/dspace/xmlworkflow/cristin/CristinFileManager.java

```diff
@@ -0,0 +1,620 @@
+            throws SQLException {
+        List<Bitstream> bss = new ArrayList<>();
+        ItemService itemService = ContentServiceFactory.getInstance().getItemService();
+        List<Bundle> bundles = itemService.getBundles(item, bundleName);
+        for (Bundle bundle : bundles) {
+            List<Bitstream> bitstreams = bundle.getBitstreams();
+            for (Bitstream bitstream : bitstreams) {
+                bss.add(bitstream);
+            }
+        }
+        return bss;
+    }
+
+    /**
+     * List all of the bitstreams which appear in the given bundle in the ORE description
+     * of the item provided by Cristin.  This will include the metadata bitstream
+     *
+     * @param doc        ORE root xml element
+     * @param bundleName
+     * @return
+     * @throws IOException
+     */
+    public List<CristinBitstream> listBitstreamsInBundle(Document doc, String bundleName)
+            throws IOException {
+        return this.listBitstreamsInBundle(doc, bundleName, true);
+    }
+
+    /**
+     * List all of the bitstreams which appear in the given bundle in the ORE description
+     * of the item provided by Cristin.  Pass in the omitMetadata flag to include or exclude
+     * the metadata bitstream.
+     *
+     * @param doc          ORE root xml element
+     * @param bundleName
+     * @param omitMetadata
+     * @return
+     * @throws IOException
+     */
+    public List<CristinBitstream> listBitstreamsInBundle(Document doc, String bundleName, boolean omitMetadata)
+            throws IOException {
+        String mdUrl = null;
+        List<Element> bitstreams = new ArrayList<Element>();
+
+        // first action is to get all the information we can from the ORE doc.
+        // This will hopefully include a link to the cristin.xml bitstream
+        // which we will need in order to learn about md5 checksums and file sequences
+        List<Element> links = this.listBitstreams(doc);
+        for (Element link : links) {
+            String incomingBundle = this.getIncomingBundleName(doc, link);
+
+            if (bundleName.equals(incomingBundle)) {
+                // this is a bitstream from the correct bundle
+                // only register it if it is not a metadata bitstream
+                boolean metadataBitstream = this.isMetadataBitstream(link.getAttributeValue("href"));
+                if (metadataBitstream) {
+                    mdUrl = link.getAttributeValue("href");
+                }
+                if (!metadataBitstream || (metadataBitstream && !omitMetadata)) {
+                    bitstreams.add(link);
+                }
+            }
+        }
+
+        // if we found a metadata bitstream, then let's get the information out of it
+        // this is where we will discover md5 checksums and file sequences
+        List<Element> fulltexts = null;
+        if (mdUrl != null) {
+            try {
+                InputStream in = this.getInputStream(mdUrl);
```

Comment:
**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *in*. Currently, there are execution paths that do not contain closure statements, e.g., when exception is thrown by SAXBuilder constructor. Either a) close *in* in a try-finally block or b) close the resource by declaring *in* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).